blog/_posts/2019-09-05-put-down-particle-accelerator.md

```diff
@@ -0,0 +1,78 @@
+
+## Development Today Needs New Tools
+For decades, the ways apps changed fit into make's model: projects got bigger, and thus build tools got faster. The move to microservices is different: it's a change in quality, not quantity.
+
+Microservices means your app isn't just one binary; your app (the experience your users care about) is the sum total of lots of individual binaries. Linking happens not at build-time but at run-time, over HTTP. Each individual build is smaller and faster. Starting the app is now more involved, as you have to connect all the components together in some way. You aren't delivering an atom; you're assembling a molecule.
+
+The new layer of working with molecules introduces new workflows. For example, `kubectl log` lets you get the independent stdout/stderr for each resource as it's running. Build tools don't automate this because it doesn't fit with their model, so it's up to the developer to make sure they're looking at the right log. It's easy to forget to check every nook and cranny, so developers end up missing an error and continue on a wild goose chase. Build tools are a hammer, and not everything's a nail.
+
+An example of a non-nail workflow: when I was at Google, I loved that our [monorepo](https://cacm.acm.org/magazines/2016/7/204032-why-google-stores-billions-of-lines-of-code-in-a-single-repository/fulltext) meant I could build and test any project with one command. It was cool that even though I never worked on GMail, I could refactor it easily. But if I wanted to actually change GMail, I'd have to go dig up a team-specific Wiki page or shell script that described how to start up the constituent pieces, spread across terminal tabs. Running your full app should be as easy as building a monolith.
+
+## Microservice Development Engine: the Chemistry Set
+Microservice development requires a new kind of tool: a Microservice Development Engine, or MDE. An MDE targets the same goal that first drove build tools: freeing you from remembering what to do after you hit save. New teammates should be able to ship code without learning the entire stack. We can't be sure exactly what an MDE will look like, but the history of build tools lets us make informed guesses.
+
+Make's superpowers came from changing the format: a build.sh script has to encode all the logic upfront, but a declarative Makefile lets the build tool engine introspect to provide a fast and consistent experience. MDEs will need a declarative description of how to build and deploy your microservice app. Because distributed systems involve the state of running services, you'll need a richer language to describe dependencies beyond the file-level data that build tools encode.
```

Comment:
> a build.sh script has to encode all the logic upfront
this might pack more punch if you mention it as the predecessor to Make up when you first introduce Make -- then this is a callback rather than a new bit of info.

Reply:
Done